laravel-app/app/Services/CommentService.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Services;
+
+use App\Http\Requests\CommentRequest;
+use App\Models\PostComment;
+use App\Models\UserPost;
+use Illuminate\Http\RedirectResponse;
+
+class CommentService
+{
+    /**
+     * Logic for storing comment
+     */
+    public function storeComment(CommentRequest $request, UserPost $post): RedirectResponse
+    {
+        $user = auth()->user();
+
+        if (!$user) {
+            return redirect()->back()->with('error', 'You must be logged in to add a comment.');
+        }
+
+        $post->comments()->create([
+            'user_id' => $user->id,
+            'comment' => $request->comment,
+        ]);
+
+        return redirect()->back()->with('success', 'Comment added successfully.');
+    }
+
+    /**
+     * Logic for deleting comment
+     */
+    public function deleteComment(int $id): RedirectResponse
+    {
+        $comment = PostComment::find($id);
+
+        if (!$comment) {
+            return redirect()->back()->with('error', 'Comment not found');
+        }
+
+        $comment->delete();
+
+        return redirect()->back()->with('success', 'Comment deleted successfully');
```

Comment:
Why does your service have redirections? what if I want to reuse the other controller and I don't want to have redirections to it??

Reply:
I'll transfer sir the redirection to the controller 🙇